crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -333,13 +341,23 @@ pub(crate) async fn bench(
                 break;
             }
         }
-    })
-    .await?;
+    });
 
-    client.stop().await;
+    while let Some(event) = segment_pipeline_event_receiver.recv().await {
+        match event {
+            SegmentPipelineEvent::Done {
+                pieces_start_index,
+                pieces_amount,
+                ..
+            } if pieces_start_index + pieces_amount >= amount_of_pieces_to_write => break,
+            _ => (),
+        }
```

Comment:
This looks smart, but two `if` branches would be more readable IMHO. I rarely find `if` in `match` to be helpful, it usually makes it harder to read.